lib/shim/specs/operation.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const RecorderSpec = require('./recorder')
+
+/**
+ * Extra parameters which may be added to an operation or query segment. All of
+ * these properties are optional.
+ *
+ * @typedef {object} DatastoreParameters
```

Comment:
this is great. we may want to file a follow up as there are other parameters types for message queues, and webframework